core/src/main/java/org/infinispan/marshall/exts/QueueExternalizers.java

```diff
@@ -0,0 +1,54 @@
+package org.infinispan.marshall.exts;
+
+import org.infinispan.commons.marshall.AbstractExternalizer;
+import org.infinispan.commons.marshall.MarshallUtil;
+import org.infinispan.commons.util.Util;
+import org.infinispan.marshall.core.Ids;
+import org.jboss.marshalling.util.IdentityIntMap;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.ArrayDeque;
+import java.util.Queue;
+import java.util.Set;
+
+public final class QueueExternalizers extends AbstractExternalizer<Queue> {
```

Comment:
I see there's a `CollectionExternalizer` already, but it only deals with `ReadOnlySegmentAwareCollection`.